pkg/remote/codec/util.go

```diff
@@ -62,6 +67,18 @@ func SetOrCheckMethodName(methodName string, message remote.Message) error {
 	return nil
 }
 
+func getServiceInfo(message remote.Message, methodName string) (svcInfo *serviceinfo.ServiceInfo, err error) {
```

Comment:
Don't getting serviceInfo by service/method here, it should be clear when client/server call invokeHandleEndpoint.

Reply:
@jayantxie I kind of agree that it should be clear to get svcInfo inside invokehandleEndpoint, but in the original implementation (L56), it needs svcInfo to check if the method name exists (`mt := svcInfo.MethodInfo(methodName)`), which I think it fits the method name `SetOrCheckMethodName`.
So I guess it's also clear that we get svcInfo here. What do you think?

Reply:
For server side, I think we could remove svc info logic from SetOrCheckMethodName to TransPipeline.Read